### ISSUE TITLE
about can_bus?

### ISSUE THREAD

**Issue by zhjw0927** [OpenDriveLab/TopoNet#21]
hello, What is the meaning of the variable 'can_bus' name?

In actual operation, the following phenomena are observed. After the four properties of the quaternion are assigned to the array, only the first property value is actually copied:

```
can_bus = np.zeros(18)
rotation = Quaternion._from_matrix(np.array(info['pose']['rotation']))
can_bus[:3] = info['pose']['translation']
can_bus[3:7] = rotation 

print(rotation) ------> Quaternion(0.7405030732851444, 0.004200992255077698, 0.01887259332437053, -0.6717747951066204)

print(can_bus) -------> array([6.65756166e+03, 1.83486642e+03, 5.96497427e+01, 7.40503073e-01,
       7.40503073e-01, 7.40503073e-01, 7.40503073e-01, 0.00000000e+00,
       0.00000000e+00, 0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
       0.00000000e+00, 0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
       0.00000000e+00, 0.00000000e+00])
```

Is this expected? or Is it pyquaternion library version issue? Thank you!

**Comment by sephyli** [comment]
Good catch! It's not expected. But fortunately it is not used in the current model.在 2024年10月11日，17:15，zhjw0927 ***@***.***> 写道：﻿
hello, What is the meaning of the variable 'can_bus' name?
In actual operation, the following phenomena are observed. After the four properties of the quaternion are assigned to the array, only the first property value is actually copied:
can_bus = np.zeros(18)
rotation = Quaternion._from_matrix(np.array(info['pose']['rotation']))
can_bus[:3] = info['pose']['translation']
can_bus[3:7] = rotation 

print(rotation) ------> Quaternion(0.7405030732851444, 0.004200992255077698, 0.01887259332437053, -0.6717747951066204)

print(can_bus) -------> array([6.65756166e+03, 1.83486642e+03, 5.96497427e+01, 7.40503073e-01,
       7.40503073e-01, 7.40503073e-01, 7.40503073e-01, 0.00000000e+00,
       0.00000000e+00, 0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
       0.00000000e+00, 0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
       0.00000000e+00, 0.00000000e+00])

Is this expected? Thank you!

**Comment by zhjw0927** [comment]
hello, please one more question. OpenLaneV2 provides only the image domain gt of the traffic element. Is this label sufficient for the task of topo relationship prediction, or is there no way to make BEV position based on argoverse data？
Thank you!

**Comment by sephyli** [comment]
The can_bus rensor is used as the can_bus encoder input. I have to keep this bug for reproducing results. Thanks again for your finding!

**Comment by sephyli** [comment]
> hello, please one more question. OpenLaneV2 provides only the image domain gt of the traffic element. Is this label sufficient for the task of topo relationship prediction, or is there no way to make BEV position based on argoverse data？ Thank you!

Labeling traffic elements in 3D space is challenging and requires significant human resources. While it is possible to label elements on the road using the intensity properties of LiDAR data, overhead traffic elements such as traffic lights or signs have sparse LiDAR points, making them difficult to label in 3D.
